### ISSUE TITLE
Missing DataGrid in the Figma set of components

### ISSUE THREAD

**Issue by EdmundsEcho** [mui-org/material-ui-x#1426]
I purchased the MUI-Figma set of components.  It's been great to use as I circle back to a design that I need to work out before going back to the code itself.  

The ui-x page includes a link to the figma set of components... I looked around but could not find the components in the v4.13 (last updated Jan 12, 2021).  Am I not looking in the right place?  

Thanks in advance for letting me know.

\- E

**Comment by oliviertassinari** [comment]
Thanks for the feedback. It's the 3rd time we get this one, only the core components have been covered, so far in Figma.  

> The ui-x page includes a link to the figma set of components...

We should remove it. I wasn't aware it existed. Could you point its location out?

**Comment by EdmundsEcho** [comment]
Great.  Here is the [link](https://material-ui.com/components/data-grid/#mit-version), go to the Resources section.  The cleanest solution would likely be to remove the link to reflect that DataGrid is not part of the core.  Otherwise, I'm not sure how to qualify the link without creating the expectation that there is something to find.

**Comment by oliviertassinari** [comment]
Thanks for the reference! Yeah, agree for removing them. in v5, we can document them with this approach once they are available:

<img width="694" alt="Screenshot 2021-04-16 at 16 41 11" src="https://user-images.githubusercontent.com/3165635/115041244-941a3b80-9ed2-11eb-91f0-d18b4e5c36d9.png">

https://next.material-ui.com/components/autocomplete/

While you are here. Which aspect of the DataGrid would you like to see covered in Figma? The UX/UI is not permanent, we still plan to improve it. The table components are relatively close to the data grid.

**Comment by EdmundsEcho** [comment]
I'm first and foremost a developer.  That means I look up the css "on the fly" and use FireFox devtools. I have a lot of respect and appreciation for what it takes to build a useful/productive/clear/pleasant UI.  I've been using MUI for 3-4 years.  I use a custom theme with a good amount in the `override` section.  

I bounce in and out of the demos and API regularly when working on this part of the project.  The design/ease-of-use in the docs is solid.  The gap is understandable, but clear and present; it only scratches the surface.

The Figma package is a way to help me navigate the best way to style the components.  Of course, how to compose what components is important, but that is mostly "table-stakes"; granted a lot of time is saved by using the same core components.

The value-add and problem Figma (and Sketch of course), solve, is how to navigate the many layers of `classNames` and css attributes.  Go too deep and you end up with an inconsistent look and feel.  Don't go deep enough, no effect.  And of course, just plain "miss" the target by no knowing about "the slot" or otherwise misspelled it, are also places where Figma is expected to help.  

At a high level, I'm always surprised to see the gap between how the expert UI folks work with MUI compared to what I was able to garner from the docs.  The Figma package is a way to help me close that gap.

Make sense?  

Feel free to ask for follow-up questions.

**Comment by oliviertassinari** [comment]
> I think MUI has a unique opportunity to build a plugin that outputs one thing: the overrides prop. Done.

@EdmundsEcho We had considered doing this for the theme. We could take the customized design tokens in the Figma kit and export a JavaScript theme (and the opposite).

@Miniplop is exploring a bit how we can output React composed code from the composition of our Figma components.

Regarding your specific proposal, it's more at the design system level, and closer to what https://modulz.app/ is trying to achieve. It could make sense for Material-UI to bring in an engineer on the Figma kit to push further the sync between design and code (at the design system level, not the composition one). So far it's been working on by one/two designers, no JavaScript engineers.

**Comment by EdmundsEcho** [comment]
@oliviertassinari I took a brief look at the Figma API.   Only to get a sense for myself, 
my quick guess at how to facilitate the process would be 

1. Name the MUI components with the MUI standard format; e.g., `MUITypography`.  This way the plugin knows where to stop/find the core components.

2. Create a map `MUI component -> attributes required from Figma`.  e.g., `MUITypography -> children.find(node => node.type === TextNode).textStyleId` (... or the like)

... does the first version have to include the selected properties for a given variant? I ask given how the overrides can depend on a specific variant.  If so, is it just a matter of converting the Figma naming convention for variants to the MUI style name?

I took a quick look at modulz.  It looks like another version of Figma except geared from the get-go to output to something like React.  It seems to do so by pre-specifying a range of core components. When I said "MUI might have a unique opportunity", I believe this is the case because the state the plugin would need to read is simplified if you narrow the range of inputs, and avoid having to go to every leaf in the Figma document (using the MUI tag).

Yes?
